integration-tests/applications/src/test/java/com/xsk/integration/tests/applications/deployment/client/XSKHttpClient.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.xsk.integration.tests.applications.deployment.client;
+
+
+import com.xsk.integration.tests.applications.Deployment;
+import com.xsk.integration.tests.applications.deployment.DeploymentException;
+import com.xsk.integration.tests.applications.kyma.TokenProvider;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.concurrent.FutureCallback;
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+import org.apache.http.message.BasicHeader;
+
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+
+import static com.xsk.integration.tests.applications.Deployment.KYMA;
+import static com.xsk.integration.tests.applications.Deployment.LOCAL;
+
+
+public class XSKHttpClient {
+
+    private CloseableHttpAsyncClient httpClient;
+
+    public XSKHttpClient(Deployment deployment) {
```

Comment:
It's probably better to have a `create()` method instead of a constructor as it's rather not cool for a constructor to make HTTP requests like the one for the Kyma tokens. Have in mind my other comment for performing a login if in the Deployment.LOCAL scenario